src/kr2_c/comparer.c

```diff
@@ -0,0 +1,35 @@
+// Успела только склеивание числа и перевод в десятичную, закинула в репо, чтобы не пропадало
+
+#include <stdio.h>
+#include <stdbool.h>
+
+int numberGetter(const bool *a, int lenA) {
+    unsigned int number = 0;
+    for (int i = 0; i < lenA; i++) {
+        number = (number << 1) | (a[i] & 1);  // сдвиг + добавление бита
+
+        // Работу кода можно объяснить в десятичных числах:
+        // Мы собираем число посимвольно из массива
+        // То есть был бы у нас массив {1,2,3},
+        // то шаги были бы такими:
+        // 1) 0*10 + 1 = 1 (первая цифра из массива)
+        // 2) 1*10 + 2 = 12 (второе цифра добавилась)
+        // 3) 12*10 + 3 = 123 (готово)
+        //
+        // А так как у нас не десятичная,
+        // мы сдвигаем биты влево (<<) и делаем |
+        // Вот только в двоичной << это *2, а не *10
+        // А | это +1 или +0 (зависимо от ситуации)
+    }
+    return number;
+}
+
+
+int main() {
+    bool arr[] = {1, 0, 1}; // 5 в десятичной (первоначальное число)
+    int lenA = 3; // длина массива а
```

Comment:
На будущее, если задаёте тестовые данные в коде, старайтесь делать так, чтобы их было легко подменить:
```suggestion
    int lenA = sizeof(arr) / sizeof(bool);
```